src/main/java/seedu/duke/parsers/AddParser.java

```diff
@@ -15,19 +15,22 @@
     private static final String FLAG = "flag";
     private static final String TASK_NAME = "taskName";
     private static final String TASK_DESCRIPTION = "taskDescription";
+    private static final String TASK_DESCRIPTION_TWO = "taskDescription2";
     private static final String TASK_WORKING_TIME = "estimatedWorkingTime";
     private static final String MODULE_CODE = "moduleCode";
     private static final String MODULE_DESCRIPTION = "moduleDescription";
+    private static final String INVALID = "invalid";
 
     // Unescaped regex for testing (split into two lines):
-    // \s*(\/t\s+(?<taskName>.+?(?=\s+-d\s+|\s+-t\s+|$))(\s+(-d\s+\"(?<taskDescription>([^\"]*))\")(?=(\s+-t\s+)|$))?
-    // (\s+(-t\s+\"(?<estimatedWorkingTime>([^\"]*))\")(?=(\s+-d\s+)|$))?|\/m\s+(?<moduleCode>\w+?(?=(\s+-d\s+)|$))
-    // (\s+(-d\s+\"(?<moduleDescription>.+)\"))?)
+    // \s*((\/t\s+(?<taskName>.+?(?=\s+-d\s+|\s+-t\s+|$))(\s+(-d\s+\"(?<taskDescription>([^\"]*))\")(?=(\s+-t\s+)|$))?
+    // (\s+(-t\s+\"(?<estimatedWorkingTime>([^\"]*))\")(?=(\s+-d\s+)|$))?(\s+(-d\s+\"(?<taskDescription2>([^\"]*))\"))?
+    // |\/m\s+(?<moduleCode>\w+?(?=(\s+-d\s+)|\s+.*$))(\s+(-d\s+\"(?<moduleDescription>.+)\"))?))(?<invalid>.*)
     // TODO: Add support for -mod argument when integrating Task and Module classes with one another
-    private static final String ADD_FORMAT = "\\s*(\\/t\\s+(?<taskName>.+?(?=\\s+-d\\s+|\\s+-t\\s+|$))"
-            + "(\\s+(-d\\s+\\\"(?<taskDescription>([^\\\"]*))\\\")(?=(\\s+-t\\s+)|$))?(\\s+(-t\\s+\\\""
-            + "(?<estimatedWorkingTime>([^\\\"]*))\\\"))?|\\/m\\s+"
-            + "(?<moduleCode>\\w+?(?=(\\s+-d\\s+)|$))(\\s+(-d\\s+\\\"(?<moduleDescription>.+)\\\"))?)";
+    private static final String ADD_FORMAT = "\\s*((\\/t\\s+(?<taskName>.+?(?=\\s+-d\\s+|\\s+-t\\s+|$))(\\s+(-d\\s+\\\""
+            + "(?<taskDescription>([^\\\"]*))\\\")(?=(\\s+-t\\s+)|$))?(\\s+(-t\\s+\\\""
+            + "(?<estimatedWorkingTime>([^\\\"]*))\\\")(?=(\\s+-d\\s+)|$))?(\\s+(-d\\s+\\\""
+            + "(?<taskDescription2>([^\\\"]*))\\\"))?|\\/m\\s+(?<moduleCode>\\w+?(?=(\\s+-d\\s+)|\\s+.*$))"
+            + "(\\s+(-d\\s+\\\"(?<moduleDescription>.+)\\\"))?))(?<invalid>.*)";
```

Comment:
Regex works when adding tasks and ignores description and working time, but fails when trying to add modules using the following format `add /m MODULE_NAME`. Not sure if intentional.

Reply:
yup i fixed the regex and added junit tests for add module so should be fine